### ISSUE TITLE
Add correct accessibility features

### ISSUE THREAD

**Issue by stowball** [iVantage/angular-ivh-treeview#174]
The W3C describe in detail how a tree view component should be written to be accessible here: https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView

Here are the paraphrased missing features:

* Missing aria `role`s: `tree`, `treeitem`, `group`
* Missing aria states: `aria-expanded`, `aria-multiselectable`, `aria-selected`
* Checkboxes shouldn't be real checkboxes
* Each `<li>` should have `tabindex="-1"` unless it is the first `aria-selected` item, then its `tabindex` is `0` (this stops keyboard users from having to tab through all of the items) to get past the tree view
* Up/down should move up/down the nodes
* Left/right should expand collapse parent nodes
* Space should (de)select a node
* Enter should (de)select a node, unless it is a parent, in which case it expands/collapses

It's worth reading the aforementioned authoring guidelines to make a component that is accessible to all users.

I understand it will be quite a bit of work, but it's very important that all users, regardless of disability and device can use this control (and every control on the web).
